src/main/ank/1.2-types-and-functions.md

```diff
@@ -0,0 +1,91 @@
+```Haskell
+x :: Integer
+```
+```kotlin
+val x: BigInteger
+```
+................
+```Haskell
+f :: Bool -> Bool
+```
+```kotlin
+val f: (Boolean) -> Boolean
+```
+................
+```Haskell
+f :: Bool -> Bool
+f x = undefined
+```
+```kotlin:ank:silent
+val f: (Boolean) -> Boolean = { x -> TODO() }
+```
+................
+```Haskell
+f :: Bool -> Bool
+f = undefined
+```
+```kotlin
+val f: (Boolean) -> Boolean = TODO()
+```
+................
+```Haskell
+fact n = product [1..n]
+```
+```kotlin:ank:silent
+fun fact(n: Int): Int = (1..n).reduce { acc, i -> acc * i }
+```
+................
+```Haskell
+absurd :: Void -> a
+```
+```kotlin
+fun <A> absurd(): (Nothing) -> A
+```
+................
+```Haskell
+f44 :: () -> Integer
+f44 () = 44
+```
+```kotlin:ank:silent
+import java.math.BigInteger
+
+val f44: () -> BigInteger = { 44.toBigInteger() }
+```
+................
+```Haskell
+fInt :: Integer -> ()
+fInt x = ()
+```
+```kotlin:ank:silent
+import java.math.BigInteger
+
+val fInt: (BigInteger) -> Unit = { x -> Unit }
+```
+................
+```Haskell
+fInt :: Integer -> ()
+fInt _ = ()
+```
+```kotlin:ank:silent
+import java.math.BigInteger
+
+val fInt: (BigInteger) -> Unit = { _ -> Unit }
+```
+................
+```Haskell
+unit :: a -> ()
+unit _ = ()
+```
+```kotlin:ank:silent
+val unit: () -> Unit = { }
```

Comment:
not sure if this is the right implementation since the `Haskell` version has an `a` type arg or it should be this one suggested below:
```suggestion
fun <A> unit(): Unit = Unit
```

Reply:
Also the lambda there should be `{ Unit }` in whatever case.

Reply:
Nice catch.....

The Scala version is `def unit[A]: A => Unit = _ => ()`

so I did `fun <A> unit: (A) -> Unit = { Unit }`

should a move it to a new line to be consistent?

Reply:
I don't think there is a standard. Personally, I prefer the new line always to make the function signature more readable. The Scala version where is all in one line is hard to read and I think Kotlin users are not used to reading functions in their curried form like in Haskell where types flow left to right. In Kotlin and Java in general, people are used to distinguishing between name, args, and return type explicitly as parts of the functions since those are clearly separated via explicit syntax in the function declaration